presto-hive/src/test/java/com/facebook/presto/hive/metastore/TestCachingHiveMetastore.java

```diff
@@ -162,21 +172,23 @@ public void testGetPartitionNames()
     {
         ImmutableList<String> expectedPartitions = ImmutableList.of(TEST_PARTITION1, TEST_PARTITION2);
         assertEquals(mockClient.getAccessCount(), 0);
-        assertEquals(metastore.getPartitionNames(TEST_METASTORE_CONTEXT, TEST_DATABASE, TEST_TABLE).get(), expectedPartitions);
-        assertEquals(mockClient.getAccessCount(), 1);
-        assertEquals(metastore.getPartitionNames(TEST_METASTORE_CONTEXT, TEST_DATABASE, TEST_TABLE).get(), expectedPartitions);
+        Table table = getTableOrElseThrow(TEST_DATABASE, TEST_TABLE);
         assertEquals(mockClient.getAccessCount(), 1);
+        assertEquals(metastore.getPartitionNames(TEST_METASTORE_CONTEXT, table).get(), expectedPartitions);
+        assertEquals(mockClient.getAccessCount(), 2);
+        assertEquals(metastore.getPartitionNames(TEST_METASTORE_CONTEXT, table).get(), expectedPartitions);
+        assertEquals(mockClient.getAccessCount(), 2);
 
         metastore.flushCache();
 
-        assertEquals(metastore.getPartitionNames(TEST_METASTORE_CONTEXT, TEST_DATABASE, TEST_TABLE).get(), expectedPartitions);
-        assertEquals(mockClient.getAccessCount(), 2);
+        assertEquals(metastore.getPartitionNames(TEST_METASTORE_CONTEXT, table).get(), expectedPartitions);
+        assertEquals(mockClient.getAccessCount(), 4);
     }
 
     @Test
     public void testInvalidGetPartitionNames()
     {
-        assertEquals(metastore.getPartitionNames(TEST_METASTORE_CONTEXT, BAD_DATABASE, TEST_TABLE).get(), ImmutableList.of());
+        assertEquals(metastore.getPartitionNames(TEST_METASTORE_CONTEXT, getTableOrElseThrow(BAD_DATABASE, TEST_TABLE)).get(), ImmutableList.of());
```

Comment:
We can call getTable since we don't want to throw an error in this case if table doesn't exist.

Reply:
As per the above comment, getPartitionNames anyways has to be changed, so this comment is taken care of.